test/ibm/runtime/test_runtime_integration.py

```diff
@@ -173,6 +169,10 @@ def test_upload_program_file(self):
         program = self.provider.runtime.program(program_id)
         self.assertTrue(program)
 
+    @unittest.skipIf(
+        not os.environ.get('QISKIT_IBM_USE_STAGING_CREDENTIALS', ''),
+        "Only runs on staging"
+    )
     def test_set_visibility(self):
```

Comment:
`test_upload_program` also makes a public test program, which we need to avoid on production. There might be others (I didn't check all):

```
    def test_upload_program(self):
        """Test uploading a program."""
        max_execution_time = 3000
        is_public = True
        program_id = self._upload_program(max_execution_time=max_execution_time,
                                          is_public=is_public)
```